src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.ConcurrentAccessDetection.netcoreapp.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Common.System;
+using Xunit;
+
+namespace Generic.Dictionary
+{
+    public class DictionaryConcurrentAccessDetectionTests
+    {
+        [Fact]
+        public static void Add_DictionaryConcurrentAccessDetection_NullComparer_ValueTypeKey()
+        {
+            Thread customThread = new Thread(() =>
```

Comment:
I'm probably spacing here -- but where is the concurrent access? There is only one thread accessing it?

Reply:
i'm simulating "concurrent access" breaking internal structure as if concurrent access happened. How i can be "sure" that threads will break internals?Depends on some random factors...i cannot guarantee break in n seconds(at least in my career trying to reproduce concurrent issue is not so easy...samples work well on my machine and doesn't work on other, [PerformanceCounter](https://github.com/dotnet/corefx/pull/26475) for instance...the behaviour are different on different machine with different number of processor, different parameters to fail test), i use thread only because if for some reason in future access detection will be removed threads doesn't loop forever blocking CI. I hope I've been clear.

Reply:
Doh - of course - that was the plan and it's what you did. BTW, I wonder if you could await a Task instead of using a Thread - @ViktorHofer the Task/Thread is only to add a timeout - if we wait on a Task perhaps it won't be marshaled to another thread, so there effectively won't be a timeout?

Reply:
> Task perhaps it won't be marshaled to another thread, so there effectively won't be a timeout?

I could do that with Task.Run+Task.Delay maybe...to simulate timeout...but maybe code will be less clear.

Reply:
You could use a Task with the LongRunning option which forces the TaskScheduler to spin up a new thread.

Reply:
yes

Reply:
@ViktorHofer @danmosemsft another idea could be use Thread with LowPriority https://source.dot.net/#System.Private.CoreLib/shared/System/Threading/ThreadPriority.cs,e968ef87c207562d

Reply:
please prefer using Task over Thread if possible.

Reply:
ok i'll do!